docs/hwm/key_value/index.rst

```diff
@@ -0,0 +1,76 @@
+.. _key_value_hwm_classes:
+
+KeyValue HWM
+========
+
+.. toctree::
+    :maxdepth: 2
+    :caption: HWM classes
+    :name: key_value_hwm_classes
+
+    key_value_int_hwm
+
+What is KeyValue HWM?
+----------------------
+
+The KeyValue High Water Mark (HWM) is a specialized class designed to manage and track incremental data changes in systems where data is stored or represented as key-value pairs, such as in message queues like Kafka.
+
+Use Case
+----------------------
+
+The ``KeyValueHWM`` class is particularly beneficial in scenarios where there is a need to `incrementally <https://onetl.readthedocs.io/en/0.10.0/strategy/incremental_strategy.html>`_ upload data in an ETL process.
+
+For instance, in typical ETL processes using `Spark with Kafka <https://spark.apache.org/docs/latest/structured-streaming-kafka-integration.html>`_, data re-written entirely from all partitions in topics starting from **zero** offset. This approach can be inefficient, time-consuming and create duplicates in target. By leveraging the ``KeyValueIntHWM`` class, it becomes possible to track the last offset of data processed. This enables the ETL process to only write new data increments, significantly reducing the amount of data transferred during each run.
```

Comment:
```suggestion
For instance, in typical ETL processes using `Spark with Kafka <https://spark.apache.org/docs/latest/structured-streaming-kafka-integration.html>`_, data re-written entirely from all partitions in topics starting from **zero** offset. This approach can be inefficient, time-consuming and create duplicates in target. By leveraging the ``KeyValueIntHWM`` class, it becomes possible to track the last offset of data processed. This enables the ETL process to read only data appended to topic since previous run, significantly reducing the amount of data transferred during each run.
```